android/src/main/java/com/imumz/flutter_split/FlutterSplitPlugin.java

```diff
@@ -16,18 +32,94 @@
   /// This local reference serves to register the plugin with the Flutter Engine and unregister it
   /// when the Flutter Engine is detached from the Activity
   private MethodChannel channel;
+  private Context appContext;
+
+
+  //"52bo50e6gavj7dpma3b0l5j2vpqep0f146nj"
+  private String apikey;
+
+  //Split
+  private SplitClient client;
+
+  // error codes
+  private static String SDK_NOT_INITIALIZED = "SNI001";
+
 
   @Override
   public void onAttachedToEngine(@NonNull FlutterPluginBinding flutterPluginBinding) {
     channel = new MethodChannel(flutterPluginBinding.getBinaryMessenger(), "flutter_split");
+    this.appContext = flutterPluginBinding.getApplicationContext();
     channel.setMethodCallHandler(this);
   }
 
   @Override
   public void onMethodCall(@NonNull MethodCall call, @NonNull Result result) {
     if (call.method.equals("getPlatformVersion")) {
       result.success("Android " + android.os.Build.VERSION.RELEASE);
-    } else {
+    }else if(call.method.equals("initializeSdk")){
+
+      this.apikey = call.argument("appKey");
+      SplitClientConfig config = SplitClientConfig.builder().build();
+
+      // Create a new user key to be evaluated
+      String matchingKey = "key";
+      Key k = new Key(matchingKey);
+      // Create factory
+      SplitFactory splitFactory = null;
+      try {
+        splitFactory = SplitFactoryBuilder.build(this.apikey, k, config, this.appContext);
+      } catch (IOException e) {
+        e.printStackTrace();
+      } catch (InterruptedException e) {
+        e.printStackTrace();
+      } catch (TimeoutException e) {
+        e.printStackTrace();
+      } catch (URISyntaxException e) {
+        e.printStackTrace();
+      }
+      // Get Split Client instance
+      this.client = splitFactory.client();
+
+    }else if(call.method.equals("trackEvent")){
+      String eventType  = call.argument("eventName");
+      HashMap<String,Object> map = new HashMap<String,Object>();
+      if(this.client!=null){
+         boolean res = client.track(eventType,map);
+         if(res){
+            result.success(true);
+         }else{
+
```

Comment:
Is `result.success(false);` or something equivalent, necessary here?
Or is this else block purposefully not executing any logic?